src/file_request_handler.cc

```diff
@@ -105,16 +106,8 @@ void FileRequestHandler::getInfo(const char* filename, UpnpFileInfo* info)
 
     auto headers = std::make_unique<Headers>();
 
-    // some resources are created dynamically and not saved in the database,
-    // so we can not load such a resource for a particular item, we will have
-    // to trust the resource handler parameter
-    if (!resourceHandler.empty()) {
-        if (resourceId == CH_DEFAULT || resourceId >= obj->getResourceCount()) {
-            throw_std_runtime_error("Requested resource not found");
-        }
-
-        auto resource = obj->getResource(resourceId);
-        auto metadataHandler = getResourceMetadataHandler(obj, resourceHandler, resource);
+    if (resourceId > 0) {
```

Comment:
You should be able to use `resource->getHandlerType() != CH_DEFAULT` instead. Resource number should not be significant anymore.